clients/go/compat.go

```diff
@@ -0,0 +1,306 @@
+}
+
+func (b WrappedBackend) EstimateGas(ctx context.Context, call ethereum.CallMsg) (gas uint64, err error) {
+	header, err := b.Backend.HeaderByNumber(ctx, nil)
+	if err != nil {
+		return 0, err
+	}
+
+	blockHash := header.Hash()
+	leash := NewLeash(header.Nonce.Uint64(), header.Number.Uint64(), blockHash[:], DefaultBlockRange)
+
+	dataPack, _ := NewDataPack(b.Signer, b.ChainID.Uint64(), call.From[:], addressToByte(call.To), DefaultGasLimit, call.GasPrice, call.Value, call.Data, leash)
+	call.Data = dataPack.Encode()
+
+	return b.Backend.EstimateGas(ctx, call)
+}
+
+func (b WrappedBackend) FilterLogs(ctx context.Context, query ethereum.FilterQuery) ([]types.Log, error) {
+	return b.Backend.FilterLogs(ctx, query)
+}
+
+func (b WrappedBackend) HeaderByNumber(ctx context.Context, number *big.Int) (*types.Header, error) {
+	return b.Backend.HeaderByNumber(ctx, number)
+}
+
+func (b WrappedBackend) PendingCodeAt(ctx context.Context, account common.Address) ([]byte, error) {
+	return b.Backend.PendingCodeAt(ctx, account)
+}
+
+func (b WrappedBackend) PendingNonceAt(ctx context.Context, account common.Address) (uint64, error) {
+	return b.Backend.PendingNonceAt(ctx, account)
+}
+
+func (b WrappedBackend) SuggestGasPrice(ctx context.Context) (*big.Int, error) {
+	return b.Backend.SuggestGasPrice(ctx)
+}
+
+func (b WrappedBackend) SuggestGasTipCap(ctx context.Context) (*big.Int, error) {
+	return b.Backend.SuggestGasTipCap(ctx)
+}
+
+func (b WrappedBackend) SubscribeFilterLogs(ctx context.Context, query ethereum.FilterQuery, ch chan<- types.Log) (ethereum.Subscription, error) {
+	return b.Backend.SubscribeFilterLogs(ctx, query, ch)
+}
+
+func NewSigner(signerFn func(digest [32]byte) ([]byte, error)) WrappedSigner {
+	return WrappedSigner{
+		SignFn: signerFn,
+	}
+}
+
+// NewKeyedTransactorWithChainID is a utility method to easily create a transaction signer
+// from a single private key.
+func NewKeyedTransactorWithChainID(key *ecdsa.PrivateKey, chainID *big.Int) (*bind.TransactOpts, error) {
+	if chainID == nil {
+		return nil, bind.ErrNoChainID
+	}
+	signer := types.LatestSignerForChainID(chainID)
+	from := crypto.PubkeyToAddress(key.PublicKey)
+	return &bind.TransactOpts{
+		GasPrice: big.NewInt(DefaultGasPrice),
+		GasLimit: DefaultGasLimit,
+		From:     from,
+		Signer: func(address common.Address, tx *types.Transaction) (*types.Transaction, error) {
+			if address != from {
+				return nil, bind.ErrNotAuthorized
+			}
+			signature, err := crypto.Sign(signer.Hash(tx).Bytes(), key)
+			if err != nil {
+				return nil, err
+			}
+			return tx.WithSignature(signer, signature)
+		},
+		Context: context.Background(),
+	}, nil
+}
+
+// NewKeyStoreTransactorWithChainID is a utility method to easily create a transaction signer from
+// an decrypted key from a keystore.
+func NewKeyStoreTransactorWithChainID(keystore *keystore.KeyStore, account accounts.Account, chainID *big.Int) (*bind.TransactOpts, error) {
```

Comment:
No. This is a comparable [wrapper](https://github.com/ethereum/go-ethereum/blob/c1aa1db69e74c71f251fc83cf7c120b4d0222728/accounts/abi/bind/auth.go#L105-L117) to the other methods.

Hmm, I think it may be possible to also remove these and code in the `GasPrice` and `GasLimit`.

Reply:
Removed. Looks okay.